web-components/icon/icon.svelte

```diff
@@ -0,0 +1,56 @@
+<svelte:options tag="leo-icon" />
+
+<script context="module" lang="ts">
+  let iconBasePath = '/icons'
+  export const setIconBasePath = (basePath: string) => (iconBasePath = basePath)
+</script>
+
+<script lang="ts">
+  export let name: string = undefined
+  let hasColor = name?.endsWith('-color')
+</script>
+
+<div class="leoIcon" part="container">
+  <slot>
+    <div
+      class="icon"
+      class:color={hasColor}
+      style:--icon-url={`url('${iconBasePath}/${name}.svg')`}
+    />
+  </slot>
+</div>
+
+<style lang="scss">
+  .leoIcon {
+    --icon-width: var(--leo-icon-size, 24px);
+    --icon-height: var(--leo-icon-size, 24px);
+    --icon-color: var(--leo-icon-color, var(--color-icon-default));
+
+    width: var(--icon-width);
+    height: var(--icon-height);
+
+    /* Slotted SVGs can use fill="currentColor" to use the icon color */
+    color: var(--leo-icon-color, inherit);
+
+    & .icon, svg {
+      width: 100%;
+      height: 100%;
+    }
+
+    /* Non color icons are set via a mask-image, so we can change the color
+     * by setting the background */
+    & .icon:not(.color) {
+      background: var(--icon-color);
```

Comment:
How about we use `background-color: currentColor` and then we can just rely on color cascading for both inline and background variants?

Reply:
I had no idea you could do this :O I thought `currentColor` was a special SVG thing ! Awesome!